### PR TITLE
refactor: icons functions, fix: open-source app icons not visible

### PR DESCRIPTION
**Pull request by ma-zal**
Changes:

fix: open-source icons apps was not visible

refactor: icon functions

### CODE REVIEW

File: src/libs/app-icon.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+		// Invert icon
+		try {
+			await invertPngAsync(dest);
+		} catch (err: any) {
+			log('error', `Failed to invert icon with URI ${dest}. Original error: ${err.message}`);
+		}
+
+		// Generate public icon
+		if (app.public && !isAppOpensource) {
```
Comment (ma-zal) [inline src/libs/app-icon.ts:60]: Here was the bug in previous implementation (code moved from AppsProvider), where Opensource apps icons was missing.

Root cause: It tried to use ".public.png" icon file for Opensource apps, but public file was not exist for opensource apps.

File: src/tree/App.js [hunk 2]
```diff
@@ -12,42 +18,56 @@ class App extends EnhancedTreeItem {
...
-		if (!this.public) {
+
+	/**
+	 * Get path to icon file.
+	 * If appp is public, then used public icon (it means icon with small green square in right bottom corner).
+	 * @param {string} iconDir Basedir of icons.
+	 * @returns {string}
+	 */
+	makeIconPath() {
+		if (!this.public || this.isOpensource) {
```
Comment (ma-zal) [inline src/tree/App.js:46]: Here was the bug in previous implementation (code moved from AppsProvider), where Opensource apps icons was missing.

Root cause: It tried to use ".public.png" icon file for Opensource apps, but public file was not exist for opensource apps.

Solution: Added `|| this.isOpensource` into condition.

File: src/libs/app-icon.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import * as Meta from "../Meta";
+import { Environment } from "../types/environment.types";
+import Jimp from "jimp";
+import { log } from "../output-channel";
+import { appsIconTempDir } from "../temp-dir";
+
+/**
+ * Downloads PNG icon from Make API and stores it in local temp dir.
+ * Also creates the inverted icon and (optionally) public icon.
+ * @returns Icon version. Is is used for icon filemame postfix.
```
Comment (mikulassychra) [inline src/libs/app-icon.ts:13]: Is is used

File: src/providers/OpensourceProvider.js [hunk 4]
```diff
@@ -210,6 +172,7 @@ class AppsProvider {
 								new Code(`expect`, "Mappable parameters", element, "imljson", Core.pathDeterminer(this._environment.version, 'module'), true)
 							]
 					}
+					break;
```
Comment (mikulassychra) [inline src/providers/OpensourceProvider.js:175]: wondering how the code could work before, without this line :)

**Review by mikulassychra — APPROVED** [review]

**Pull request merged**